tests/neg-custom-args/fatal-warnings/i2673c.scala

```diff
@@ -0,0 +1,6 @@
+package Foos
+
+object Outer {
+  class X // error
+  object x
```

Comment:
@nicolasstucki Why is this not allowed? It is only an issue when `class X` and `object x` are top level, right? Scalac does not complain

Reply:
Those classes will emit `Outer$X.class` and `Outer$x.class` which may conflict.

Reply:
I don't think so. I think `Outer$x$.class` will be emitted. So there is no conflict. I think it is only an issue for top level objects

Reply:
Yes, the object will generate `Outer$x$.class` but is the class needs a companion object it will generate `Outer$X$.class` which causes the conflict.

Reply:
That would happen with
```scala
object Outer {
  case class X()
  object x
}
```
or any other class that generates the module class for it.